hbase-client/src/main/java/org/apache/hadoop/hbase/exceptions/ClientExceptionsUtil.java

```diff
@@ -87,15 +83,15 @@ public static Throwable findException(Object exception) {
     }
     Throwable cur = (Throwable) exception;
     while (cur != null) {
-      if (isSpecialException(cur)) {
+      if (regionDefinitelyOnTheRegionServerException(cur)) {
```

Comment:
I think here we still need the isSpecialException? Otherwise we may go into the second condition where we get the cause of the exception for some exceptions which are not in regionDefinitelyOnTheRegionServerException but still in isSpecialException.
Of cource I do not think 'isSpecialException' is a good name, maybe we could find another name for it, and makes it private.